scipy/stats/_continuous_distns.py

```diff
@@ -3569,34 +3569,61 @@ def fit(self, data, *args, **kwds):
         data, floc, fscale = _check_fit_input_parameters(self, data,
                                                          args, kwds)
 
-        # if user has provided `floc` or `fscale`, fall back on super fit
-        # method. This scenario is not suitable for solving a system of
-        # equations
-        if floc is not None or fscale is not None:
-            return super().fit(data, *args, **kwds)
-
-        # rv_continuous provided guesses
-        loc, scale = self._fitstart(data)
-        # account for user provided guesses
-        loc = kwds.pop('loc', loc)
-        scale = kwds.pop('scale', scale)
-
         # By the method of maximum likelihood, the estimators of the
         # location and scale are the roots of the equation defined in
         # `func` and the value of the expression for `loc` that follows.
         # Source: Statistical Distributions, 3rd Edition. Evans, Hastings,
         # and Peacock (2000), Page 101
 
-        def func(scale, data):
-            sdata = -data / scale
-            wavg = _average_with_log_weights(data, logweights=sdata)
-            return data.mean() - wavg - scale
+        def get_loc_from_scale(scale):
+            return -scale * (sc.logsumexp(-data / scale) - np.log(len(data)))
+
+        if fscale is not None:
+            # if the scale is fixed, the location can be analytically
+            # determined.
+            scale = fscale
+            loc = get_loc_from_scale(scale)
+
+        else:
+            # A different function is solved depending on whether the location
+            # is fixed.
+            if floc is not None:
+                # equation to use if the location is fixed.
+
+                def func(scale):
+                    return (((floc - data) * np.exp(
+                        (floc - data) / scale) + data).sum() -
+                            len(data) * (floc + scale))
```

Comment:
Can this be broken into intermediate variables like the `func` below, with each fitting on a line? Something like (but check me):
```suggestion
                    term1 = (floc - data) * np.exp((floc - data) / scale)
                    term2 = len(data) * (floc + scale)
                    return term1.sum() + data.sum() + term2
```

Reply:
Should be resolved.